File: BankingSystemFullUITest.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;

public class BankingSystemFullUITest {
    private WebDriver driver;

    @Before
    public void setUp() {
        // Setup ChromeDriver
        driver = new ChromeDriver();
        // Provide the absolute path to your dashboard HTML provided in the artifacts
        driver.get("file:///path/to/your/banking_dashboard.html");
    }

    @Test
    public void testVerifiedStateFunctionality() {
        
        String status = driver.findElement(By.id("status-label")).getText();
        
        if (status.equals("Verified")) {
            assertTrue("Deposit button should be enabled", driver.findElement(By.id("deposit-btn")).isEnabled());
            assertTrue("Withdraw button should be enabled", driver.findElement(By.id("withdraw-btn")).isEnabled());
            assertTrue("Transfer button should be enabled", driver.findElement(By.id("transfer-btn")).isEnabled());
        }
    }

    @Test
    public void testSuspendedStateRestrictions() {
        
        WebElement statusLabel = driver.findElement(By.id("status-label"));
        
        if (statusLabel.getText().contains("Suspended")) {
            WebElement transferBtn = driver.findElement(By.id("transfer-btn"));
            WebElement withdrawBtn = driver.findElement(By.id("withdraw-btn"));
            
            
            assertFalse("Transfer should be disabled when Suspended", transferBtn.isEnabled());
            assertFalse("Withdraw should be disabled when Suspended", withdrawBtn.isEnabled());
        }
    }

    @Test
    public void testClosedStateLockdown() {
       
        WebElement statusLabel = driver.findElement(By.id("status-label"));
        
        if (statusLabel.getText().contains("Closed")) {
            WebElement depositBtn = driver.findElement(By.id("deposit-btn"));
            WebElement withdrawBtn = driver.findElement(By.id("withdraw-btn"));
            
            
            assertFalse("Deposit blocked in Closed state", depositBtn.isEnabled());
            assertFalse("Withdraw blocked in Closed state", withdrawBtn.isEnabled());
        }
    }

    @Test
    public void testInputValidationNegativeDeposit() {
        
        WebElement depositInput = driver.findElement(By.id("amount-input"));
        WebElement depositBtn = driver.findElement(By.id("deposit-btn"));
        WebElement notification = driver.findElement(By.id("notification-box"));

        depositInput.clear();
        depositInput.sendKeys("-50");
        depositBtn.click();

        
        assertTrue("Notification box should be visible", notification.isDisplayed());
        assertTrue("Should display 'Invalid' message", notification.getText().contains("Invalid"));
    }

    @After
    public void tearDown() {
        if (driver != null) {
            driver.quit();
        }
    }
}

File: BankingSystemFullUITest.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import unittest

class BankingSystemFullUITest(unittest.TestCase):

    def setUp(self):
        self.driver = webdriver.Chrome()
        self.driver.get("file:///path/to/your/banking_dashboard.html")

    def test_verified_state_functionality(self):
        status = self.driver.find_element(By.ID, "status-label").text
        if status == "Verified":
            self.assertTrue(self.driver.find_element(By.ID, "deposit-btn").is_enabled())
            self.assertTrue(self.driver.find_element(By.ID, "withdraw-btn").is_enabled())
            self.assertTrue(self.driver.find_element(By.ID, "transfer-btn").is_enabled())

    def test_suspended_state_restrictions(self):
        status_label = self.driver.find_element(By.ID, "status-label")
        if "Suspended" in status_label.text:
            transfer_btn = self.driver.find_element(By.ID, "transfer-btn")
            withdraw_btn = self.driver.find_element(By.ID, "withdraw-btn")
            self.assertFalse(transfer_btn.is_enabled())
            self.assertFalse(withdraw_btn.is_enabled())

    def test_closed_state_lockdown(self):
        status_label = self.driver.find_element(By.ID, "status-label")
        if "Closed" in status_label.text:
            deposit_btn = self.driver.find_element(By.ID, "deposit-btn")
            withdraw_btn = self.driver.find_element(By.ID, "withdraw-btn")
            self.assertFalse(deposit_btn.is_enabled())
            self.assertFalse(withdraw_btn.is_enabled())

    def test_input_validation_negative_deposit(self):
        deposit_input = self.driver.find_element(By.ID, "amount-input")
        deposit_btn = self.driver.find_element(By.ID, "deposit-btn")
        notification = self.driver.find_element(By.ID, "notification-box")
        deposit_input.clear()
        deposit_input.send_keys("-50")
        deposit_btn.click()
        self.assertTrue(notification.is_displayed())
        self.assertIn("Invalid", notification.text)

    def tearDown(self):
        self.driver.quit()

if __name__ == "__main__":
    unittest.main()
